includes/notes/class-wc-payments-notes-set-up-stripelink.php

```diff
@@ -0,0 +1,108 @@
+<?php
+/**
+ * Set up Stripe Link note for WooCommerce inbox.
+ *
+ * @package WooCommerce\Payments\Admin
+ */
+
+use Automattic\WooCommerce\Admin\Notes\NoteTraits;
+use WCPay\Payment_Methods\Link_Payment_Method;
+use WCPay\Payment_Methods\CC_Payment_Method;
+
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * Class WC_Payments_Notes_Setup_StripeLink
+ */
+class WC_Payments_Notes_Set_Up_StripeLink {
+	use NoteTraits;
+
+	/**
+	 * Name of the note for use in the database.
+	 */
+	const NOTE_NAME = 'wc-payments-notes-set-up-stripe-link';
+
+	/**
+	 * CTA button link
+	 */
+	const NOTE_DOCUMENTATION_URL = 'https://woocommerce.com/document/payments/woocommerce-payments-stripe-link/';
+
+	/**
+	 * The account service instance.
+	 *
+	 * @var WC_Payment_Gateway_WCPay
+	 */
+	private static $gateway;
+
+	/**
+	 * Checks if a note can and should be added.
+	 *
+	 * @return bool
+	 */
+	public static function should_display_note() {
+		// If UPE is not enabled, skip.
+		if ( ! \WC_Payments_Features::is_upe_enabled() ) {
+			return false;
+		}
+
+		// Check if Link payment is available.
+		$available_upe_payment_methods = self::$gateway->get_upe_available_payment_methods();
+
+		if ( ! in_array( Link_Payment_Method::PAYMENT_METHOD_STRIPE_ID, $available_upe_payment_methods, true ) ) {
+			return false;
+		}
+
+		// Retrieve enabled payment methods at checkout.
+		$enabled_payment_methods = self::$gateway->get_payment_method_ids_enabled_at_checkout( null, true );
+
+		// If card payment method is not enabled, skip.
+		if ( ! in_array( CC_Payment_Method::PAYMENT_METHOD_STRIPE_ID, $enabled_payment_methods, true ) ) {
```

Comment:
To keep related pieces together, can we merge this and next if-statements into one, please?

Reply:
Done